### PR TITLE
syscall.c: Redo openDatabase and rename it to load_sdb

### PR DESCRIPTION
**Pull request by kazarmy**
**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [X] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

This pr redoes `syscall.c`'s `openDatabase()` according to https://github.com/rizinorg/rizin/pull/379#issuecomment-762002283 and renames it to `load_sdb()`.

**Test plan**

All builds should be green.

**Closing issues**

...

### CODE REVIEW

**Review by XVilka — APPROVED** [review]

File: librz/syscall/syscall.c [hunk 1]
```diff
@@ -42,22 +42,22 @@ RZ_API void rz_syscall_free(RzSyscall *s) {
...
+			sdb_reset(*db);
+			sdb_open(*db, file);
 		} else {
-			db = sdb_new(0, file, 0);
+			*db = sdb_new(0, file, 0);
 		}
-	} else {
-		sdb_free(db);
-		db = sdb_new0();
+		free(file);
```
Comment (XVilka) [inline librz/syscall/syscall.c:56]: Shouldn't this be the opposite?

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/syscall/syscall.c (hunk 1 above)
Comment (kazarmy) [reply librz/syscall/syscall.c:56]: It should return true if the file exists, right?

Comment (XVilka) [reply librz/syscall/syscall.c:56]: Yes, I think so.

Comment (kazarmy) [reply librz/syscall/syscall.c:56]: > It should return true if the file exists, right?

The diff is a bit confusing, but that's what the code does.

**Review by XVilka — APPROVED** [review]

**Pull request merged**